File: Exercise/exercise11/Solve1.java
public class Solve1 {
        public static void main(String... args) {
                fib(7);
        }

        public static void fib(int n) {
                int a = 0;
                int b = 1;
                int t;

                for (int i = 0; i < n; i++) {
                        t = a;
                        a = b;
                        b = a + t;
                        System.out.println(a);
                }
        }
}

File: Exercise/exercise11/solve1.c
#include <stdio.h>

void fib(int n);

int main() {

        fib(12);
        return 0;
}

void swap(int *a, int *b) {
        int t = *a;
        *a = *b;
        *b = t;
}

void fib(int n) {
        int a, b;
        a = 0;
        b = 1;

        for (int i = 0; i < n; i++) {
                swap(&a, &b);
                a = a + b;
                printf("%d ", a);
        }
}
